packages/block-editor/src/components/inserter/hooks/use-block-types-state.js

```diff
@@ -23,20 +23,18 @@ import { store as blockEditorStore } from '../../../store';
  * @return {Array} Returns the block types state. (block types, categories, collections, onSelect handler)
  */
 const useBlockTypesState = ( rootClientId, onInsert ) => {
-	const { categories, collections, items } = useSelect(
-		( select ) => {
-			const { getInserterItems } = select( blockEditorStore );
-			const { getCategories, getCollections } = select( blocksStore );
-
-			return {
-				categories: getCategories(),
-				collections: getCollections(),
-				items: getInserterItems( rootClientId ),
-			};
-		},
+	const [ items ] = useSelect(
+		( select ) => [
+			select( blockEditorStore ).getInserterItems( rootClientId ),
+		],
```

Comment:
Nit: I think we can just return `items` here as we're dealing with a single collection now.

Reply:
Then `useSelect` would shallow compare all the elements on each call, and that's wasted time when it can just compare `oldItems === newItems` once, on the memoized array.

Reply:
> I thought it would bail out early if `oldItems === newItems` and avoid shallow comparison.

@mamaduka you're right! `@wordpress/is-shallow-equal` really does a `===` comparison first and returns early.

But I think I'll keep the `[ items ]` form anyway, for the sake of consistency 🙂